fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1850,6 +1858,9 @@ private Key.Evaluated deleteRecordsWhereCheckRecordTypes() {
                 return null;
             }
 
+            // Note: it's possible that the RecordCount is DISABLED, and we could do the deleteWhere, but in a world
+            // where some stores may have it disabled, and others may not, it's probably best to keep this code
+            // simple
```

Comment:
I'm not sure I buy this argument, as it doesn't seem like the code editions would be that different. We already look at the index state when selecting the set of `IndexMaintainer`s that we need to update, and it seems like we should mirror that behavior here. I could also see the argument that for this function, we should always look at the index, so that this function fails whenever the record meta-data is updated, rather than waiting for the index to be built, but that seems like a separate change.

Reply:
I think that's a fairly solid point.
I also, realized, that if someone is changing something around the RecordCountState, they would probably look for usages, to double check what they need to update, if we're not referencing it here, that could be missed.

I would like to fix this as part of this PR.